skimage/filters/rank/tests/test_rank.py

```diff
@@ -104,6 +112,14 @@ def check():
                 result[19, 18] = 172
                 assert_array_almost_equal(expected, result)
             else:
+                if outdt is not None:
+                    # Avoid rounding issues comparing to expected result
+                    if filter == 'sum':
+                        # sum test data seems to be 8-bit disguised as 16-bit
+                        datadt = np.uint8
+                    else:
+                        datadt = expected.dtype
+                    result = result.astype(datadt)
```

Comment:
```suggestion
                    result = result.astype(expected.dtype)
```

It seems the 2D data for 'sum' is already uint8, and it is only the 3D case that needed this special handling.

Reply:
Thanks, yes, that simplifies things